Core/CoreUITests/Discussions/DiscussionReplyTests.swift

```diff
@@ -30,25 +30,23 @@ class DiscussionReplyTests: CoreUITestCase {
             create_discussion_topic: true
     )))
 
-    func mockDiscussion(allowAttachment: Bool? = true, allowRating: Bool = false) -> APIDiscussionTopic {
-        let discussId: ID = "1"
-        let discussion = APIDiscussionTopic.make(
-            id: discussId,
-            message: "top message",
-            html_url: URL(string: "/courses/\(course.id)/discussion_topics/\(discussId)"),
-            permissions: .make(attach: allowAttachment),
-            allow_rating: allowRating
-        )
+    var markedAsRead: [ID: Bool] = [:]
+
+    @discardableResult
+    func mockDiscussion(_ discussion: APIDiscussionTopic = .make(), fullTopic: APIDiscussionFullTopic = .make()) -> APIDiscussionTopic {
         mockData(ListDiscussionTopicsRequest(context: course), value: [discussion])
-        mockData(GetTopicRequests(context: course, topicID: discussId.value), value: discussion)
-        let fullTopic = APIDiscussionFullTopic.make()
-        mockData(GetFullTopicRequests(context: course, topicID: discussId.value), value: fullTopic)
-        mockData(ListDiscussionEntriesRequest(context: course, topicID: discussId.value), value: fullTopic.view)
-        let readDiscussionUrl = URL(string: "https://canvas.instructure.com/api/v1/courses/\(course.id)/discussion_topics/\(discussId)/read")!
+        mockData(GetTopicRequests(context: course, topicID: discussion.id.value), value: discussion)
+        mockData(GetFullTopicRequests(context: course, topicID: discussion.id.value), value: fullTopic)
+        mockData(ListDiscussionEntriesRequest(context: course, topicID: discussion.id.value), value: fullTopic.view)
+        let readDiscussionUrl = URL(string: "https://canvas.instructure.com/api/v1/courses/\(course.id)/discussion_topics/\(discussion.id)/read")!
         mockURL(readDiscussionUrl, response: HTTPURLResponse(url: readDiscussionUrl, statusCode: 204, httpVersion: nil, headerFields: [:]))
-        fullTopic.unread_entries.forEach {
-            let url = URL(string: "https://canvas.instructure.com/api/v1/courses/\(course.id)/discussion_topics/\(discussId)/entries/\($0)/read")!
-            mockURL(url, response: HTTPURLResponse(url: url, statusCode: 204, httpVersion: nil, headerFields: [:]))
+        fullTopic.unread_entries.forEach { entry in
+            let url = URL(string: "https://canvas.instructure.com/api/v1/courses/\(course.id)/discussion_topics/\(discussion.id)/entries/\(entry)/read")!
+            mockResponse(URLRequest(url: url)) { [weak self] _ in
+                self?.markedAsRead[entry] = true
+                print("entry \(entry) marked as read")
```

Comment:
Probably don't want to check-in the print right?

Reply:
Indeed